scripts/west_commands/build.py

```diff
@@ -373,6 +373,22 @@ def _run_cmake(self, cmake_opts):
             final_cmake_args.extend(cmake_opts)
         cmake.run_cmake(final_cmake_args)
 
+    def _run_pristine(self):
+        log.inf('Making build dir {} pristine'.format(self.build_dir))
+
+        zb = os.environ.get('ZEPHYR_BASE')
+        if not zb:
+            log.die("Internal error: os.environ.ZEPHYR_BASE not set")
+
+        if not is_zephyr_build(self.build_dir):
+            log.die('Refusing to run pristine on a folder that is not a Zephyr '
+                    'build system')
+
+        cmake_args = ['-P', '{}/cmake/pristine.cmake'.format(zb)]
+        cmake = shutil.which('cmake')
+        cmd = [cmake] + cmake_args
+        subprocess.check_call(cmd, cwd=self.build_dir)
```

Comment:
`west.cmake.run_cmake` is a little safer. It prints a sensible error message if `shutil.which('cmake')` returns `None.` Might want to switch to it or replicate the error handling here.